File: charactermanager/backend/src/main/java/ai/ragu/charactermanager/enumeration/ActivationEnum.java
package ai.ragu.charactermanager.enumeration;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

import java.util.HashMap;
import java.util.Map;

public enum ActivationEnum {
    INTERACT("interact"),
    STRIKE("strike"),
    ONE_ACTION("oneAction"),
    ONE_OR_MORE_DAYS("oneOrMoreDays"),
    BULWARK("bulwark");

    private final String value;
    private final static Map<String, ActivationEnum> CONSTANTS = new HashMap<>();

    static {
        for (ActivationEnum c: values()) {
            CONSTANTS.put(c.value, c);
        }
    }

    ActivationEnum(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return this.value;
    }

    @JsonValue
    public String value() {
        return this.value;
    }

    @JsonCreator
    public static ActivationEnum fromValue(String value) {
        ActivationEnum constant = CONSTANTS.get(value);
        if (constant == null) {
            throw new IllegalArgumentException(value);
        } else {
            return constant;
        }
    }
}

File: charactermanager/backend/src/main/java/ai/ragu/charactermanager/enumeration/ActivationEnum.kt
package ai.ragu.charactermanager.enumeration

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue
import java.util.*

enum class ActivationEnum(private val value: String) {
    INTERACT("interact"),
    STRIKE("strike"),
    ONE_ACTION("oneAction"),
    ONE_OR_MORE_DAYS("oneOrMoreDays"),
    BULWARK("bulwark");

    companion object {
        private val CONSTANTS: MutableMap<String, ActivationEnum> = HashMap()

        @JsonCreator
        fun fromValue(value: String?): ActivationEnum {
            val constant = CONSTANTS[value]
            return constant ?: throw IllegalArgumentException(value)
        }

        init {
            for (c in values()) {
                CONSTANTS[c.value] = c
            }
        }
    }

    override fun toString(): String {
        return value
    }

    @JsonValue
    fun value(): String {
        return value
    }

}